cmd/otel-allocator/server/server.go

```diff
@@ -64,12 +64,16 @@ func NewServer(log logr.Logger, allocator allocation.Allocator, discoveryManager
 		discoveryManager: discoveryManager,
 		compareHash:      uint64(0),
 	}
-	router := mux.NewRouter().UseEncodedPath()
+
+	router := gin.Default()
+	router.UseRawPath = true
+	router.UnescapePathValues = false
```

Comment:
Why is it explicit set to false?

Reply:
It is set to `true` by default so I'm setting it to `false` explicitly here. Basically these two options should provide same functionality as we had with `gorilla/mux` and [`UseEncodedPath`](https://pkg.go.dev/github.com/gorilla/mux#Router.UseEncodedPath). 

If it helps I can add a comment, although if you consult the comments for the [`gin.Engine`](https://pkg.go.dev/github.com/gin-gonic/gin#Engine) type it should be clearer.